### PR TITLE
Associations

### PR DESCRIPTION
**Pull request by phelian23**
In this PR
- Add book class
- Add classroom class
- Add rental class
- Add Associations between classes

### CODE REVIEW

File: book.rb [hunk 1]
```diff
@@ -0,0 +1,13 @@
+class Book
+  attr_accessor :title, :author, :rentals
+
+  def initialize(title, author, rentals = [])
+    @title = title
+    @author = author
+    @rentals = rentals
+  end
```
Comment (TedLivist) [inline book.rb]: - [ ] **[OPTIONAL]** You probably don't need to define the empty **`rentals`** in the parameters like this to accomplish the initialization. The following code snippet works too. This way there will be a lesser number of arguments when instantiating a Book object and the code looks cleaner.
  
  ```
  def initialize(title, author)
      @title = title
      @author = author
      @rentals = []
    end
  ```

File: person.rb [hunk 2]
```diff
@@ -1,15 +1,17 @@
...
+  attr_accessor :name, :age, :parent_permission, :rentals
 
-  def initialize(age, name = 'unknown', parent_permission: true)
+  def initialize(age, name = 'unknown', rentals = [], parent_permission: true)
     @id = rand(1..1000)
     @name = name
     @age = age
     @parent_permission = parent_permission
     @corrector = Corrector.new
+    @rentals = rentals
```
Comment (TedLivist) [inline person.rb]: - [ ] **[OPTIONAL]** You probably do not need to define the **`rentals`** parameter here. The `rentals` variable can also be implicitly initialized as demonstrated in the suggestion above.

**Review by TedLivist — CHANGES_REQUESTED** [review]
## STATUS: CHANGES REQUIRED ♻️ ♻️

Hi @phelian23,

Good job so far!
✔️ Good documentation
✔️ Addition of has-many-through association methods between Book and Person, through Rental 
However, there are some issues that you still need to work on to go to the next project but you are almost there!

## Required Changes ♻️
**I am requesting this change here because you made no changes to the `student.rb` file in this PR**
- [ ] Your **Student** class should have a method that serves as the `belongs-to` side of the association with the **Classroom**. Kindly add it. You can refer to this [article](https://github.com/microverseinc/curriculum-ruby/blob/main/oop/articles/oop_relationships_by_examples.md#both-ways-relationships) on how to do this.

  phelian-student

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by Emmyn5600 — APPROVED** [review]
STATUS: APPROVED :partying_face: 

Wooow👏, Your project is complete!  There is nothing else to say other than... it's time to merge it. 👏

Congratulations! 🎉 


Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.

**Pull request merged**